core/connection_test.go

```diff
@@ -0,0 +1,929 @@
+package connection_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eth"
+	rpccalls "github.com/Layr-Labs/eigensdk-go/metrics/collectors/rpc_calls"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	servicemanager "github.com/yetanotherco/aligned_layer/contracts/bindings/AlignedLayerServiceManager"
+	connection "github.com/yetanotherco/aligned_layer/core"
+	"github.com/yetanotherco/aligned_layer/core/chainio"
+	"github.com/yetanotherco/aligned_layer/core/config"
+	"github.com/yetanotherco/aligned_layer/core/utils"
+)
+
+func DummyFunction(x uint64) (uint64, error) {
+	fmt.Println("Starting Anvil on Port ")
+	if x == 42 {
+		return 0, connection.PermanentError{Inner: fmt.Errorf("Permanent error!")}
+	} else if x < 42 {
+		return 0, fmt.Errorf("Transient error!")
+	}
+	return x, nil
+}
+
+func TestRetryWithData(t *testing.T) {
+	function := func() (*uint64, error) {
+		x, err := DummyFunction(43)
+		return &x, err
+	}
+	data, err := connection.RetryWithData(function, 1000, 2, 3, connection.MaxInterval)
+	if err != nil {
+		t.Errorf("Retry error!: %s", err)
+	} else {
+		fmt.Printf("DATA: %d\n", data)
```

Comment:
I don't think we need to log data here, I would make the tests pass without logs and only log on error cases

Reply:
addressed